src/Compilers/Shared/BuildClient.cs

```diff
@@ -249,6 +255,30 @@ private static bool UseNativeArguments()
             return true;
         }
 
+        private static bool AreNamedPipesSupported()
+        {
+            if (!PlatformInformation.IsRunningOnMono)
+                return true;
+
+            IDisposable npcs = null;
+            try 
+            {
+                // Mono configurations without named pipe support will throw a PNSE at some point in this process.
+                npcs = new NamedPipeClientStream(".", "nonexistentpipe", PipeDirection.InOut);
```

Comment:
Think we should find a less common name to use for the check here. Connecting to a named pipe is an observable event. Hence shoudl minimize the chance this accidentally connects to a real pipe. Consider a generated SHA here or name prefixed with something akin to "mono-org-tools" (a name that is more likely to be mono team specific)

Reply:
Good point, will do. Random guid + a name prefix is fine then?